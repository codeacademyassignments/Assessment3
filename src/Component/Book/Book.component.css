.bookImage {
  width:  100%;
  height: 303px;
  background-repeat:   no-repeat;
  background-size:     cover;
}
.bookTitle{
  text-align: center;
  padding: 0% 9%;
}

.book{
  width:350px;
  margin: 2% 1%;
  background-color: orange;
  display: flex;
    flex-direction: column;
    justify-content: space-between;
}