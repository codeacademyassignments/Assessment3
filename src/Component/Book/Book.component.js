import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Book.component.css';
import image from '../../Images/cover_image.jpeg';
import BookFooter from './BookFooter/BookFooter.component';

class Book extends Component {
  render() {
    const styleForBook = this.props.index % 2 === 0 ? 'bookOnEvenIndex' : 'bookOnOddIndex';
    return (
      <article className={styleForBook}>
        <div className="bookImage" style={{ backgroundImage: `url(${image})` }} />
        <p className="bookTitle">{this.props.title}</p>
        <BookFooter rating={this.props.rating} bookId={this.props.bookId} liked={this.props.liked} onHeartClick={(data) => { }} index={this.props.index} />
      </article>
    );
  }
}

Book.propTypes = {
  title: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  bookId: PropTypes.number.isRequired,
  liked: PropTypes.bool,
  index: PropTypes.number,
};

Book.defaultProps = {
  liked: false,
  index: 0,
};

export default Book;
