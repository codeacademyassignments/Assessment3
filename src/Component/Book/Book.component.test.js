import React from 'react';
import renderer from 'react-test-renderer';
import Book from './Book.component';
import TestData from '../../TestData/TestData';

const blog = TestData['J K Rowling'][0];
describe('Section', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<Book rating={blog.rating} liked title={blog.Name} bookId={blog.id} />);
    expect(tree).toMatchSnapshot();
  });
});
