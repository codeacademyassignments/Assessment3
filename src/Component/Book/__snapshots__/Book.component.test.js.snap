// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section snapshot should match 1`] = `
<article
  className="bookOnEvenIndex"
>
  <div
    className="bookImage"
    style={
      Object {
        "backgroundImage": "url(cover_image.jpeg)",
      }
    }
  />
  <p
    className="bookTitle"
  >
    Harry Potter and the Sorcerers Stone (Harry Potter, #1)
  </p>
  <div
    className="cardFooter"
  >
    <div>
      <p
        className="rating"
      >
        4.45
      </p>
    </div>
    <div
      className="heartImage"
    >
      <img
        alt="heart-black"
        className="heart"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": [Function],
            "__esModule": true,
            "default": "heart-red.svg",
          }
        }
      />
    </div>
  </div>
</article>
`;
